core/exception/Exception.ts

```diff
@@ -17,96 +19,103 @@ import {isString} from 'lodash';
-     *      be treated as a serialized exception and used to construct the returned Error.
+     * @param fetchResponse - return value of native fetch, as enhanced by FetchService.
      */
-    static fetchError(fetchOptions: FetchOptions, response) {
-        const httpStatus = response.status,
+    static fetchError(fetchOptions: FetchOptions, fetchResponse: FetchResponse): FetchException {
+        const {headers, status, statusText, responseText} = fetchResponse,
             defaults = {
-                name: 'HTTP Error ' + (httpStatus || ''),
-                message: response.statusText,
-                httpStatus,
-                serverDetails: response.responseText,
+                name: 'HTTP Error ' + (status || ''),
+                httpStatus: status,
+                message: statusText,
+                serverDetails: responseText,
                 fetchOptions
             };
 
-        if (httpStatus === 401) {
-            return this.createInternal(defaults, {
+        if (status === 401) {
+            return this.createInternal({
+                ...defaults,
                 name: 'Unauthorized',
                 message: 'Your session may have timed out and you may need to log in again.'
-            });
+            }) as FetchException;
         }
 
         // Try to "smart" decode as server provided JSON Exception (with a name)
         try {
-            const cType = response.headers.get('Content-Type');
+            const cType = headers.get('Content-Type');
             if (cType && cType.includes('application/json')) {
-                const serverDetails = JSON.parse(response.responseText);
+                const serverDetails = JSON.parse(responseText);
                 if (serverDetails?.name) {
-                    return this.createInternal(defaults, {
+                    return this.createInternal({
+                        ...defaults,
                         name: serverDetails.name,
                         message: serverDetails.message,
                         isRoutine: serverDetails.isRoutine ?? false,
                         serverDetails
-                    });
+                    }) as FetchException;
                 }
             }
         } catch (ignored) {}
 
         // Fall back to raw defaults
-        return this.createInternal(defaults, {});
+        return this.createInternal(defaults) as FetchException;
     }
 
     /**
      * Create an Error to throw when a fetch call is aborted.
      * @param fetchOptions - original options the app passed to FetchService.
      * @param e - Error thrown by native fetch
      */
-    static fetchAborted(fetchOptions: FetchOptions, e) {
+    static fetchAborted(fetchOptions: FetchOptions, e: any): FetchException {
         return this.createInternal({
             name: 'Fetch Aborted',
             message: `Fetch request aborted, url: "${fetchOptions.url}"`,
             isRoutine: true,
             isFetchAborted: true,
```

Comment:
Should we add `isFetchAborted` to the FetchException interface?

Reply:
Done